test/functional/lua/base64_spec.lua

```diff
@@ -0,0 +1,72 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear = helpers.clear
+local exec_lua = helpers.exec_lua
+local eq = helpers.eq
+local pcall_err = helpers.pcall_err
+local matches = helpers.matches
+
+describe('vim.base64', function()
+  before_each(clear)
+
+  local function encode(s)
+    return exec_lua([[return vim.base64.encode(...)]], s)
+  end
+
+  local function decode(s)
+    return exec_lua([[return vim.base64.decode(...)]], s)
+  end
+
+  it('works', function()
+    local values = {
+      '',
+      'Many hands make light work.',
+      [[
+        Call me Ishmael. Some years ago—never mind how long precisely—having little or no money in
+        my purse, and nothing particular to interest me on shore, I thought I would sail about a
+        little and see the watery part of the world.
+      ]],
+      [[
+        It is a truth universally acknowledged, that a single man in possession of a good fortune,
+        must be in want of a wife.
+      ]],
+      'Happy families are all alike; every unhappy family is unhappy in its own way.',
```

Comment:
as an additional check, I would also `eq([[the base64 data]], encode(v))` at least for the shortest one.

Reply:
Added one more test case with an explicit check on the encoded output.